oauth2.md

```diff
@@ -0,0 +1,169 @@
+# OAuth 2.0 Integration
+
+This document provides the steps for partners to integrate their applications to Moca using OAuth 2.0 (Authorization Code Flow).
+
+## Steps
+
+### 1. Application Registration
+
+- Provide `redirect_uri` values to Moca
+- Provide a public key to Moca
+- Obtain Partner UUID
+
+### 2. User Authorization Request
+
+TODO: In the "Login with Moca" button, specify the following parameters:
+- `response_type`: must be set to `code`
+- `client_id`: The obtained partner UUID from step 1
+- `redirect_uri`: One of the registered `redirect_uri` from step 1
+- `state`: Must be generated by partner for security measures
+
+If the `client_id` and `redirect_uri` parameters are correct, any errors that occur will be sent via HTTP 302 redirect to the specified `redirect_uri`.
```

Comment:
did you mean incorrect?

Reply:
Yes, because if one of those (`client_id` and/or `redirect_uri`) are not valid, instead redirecting the error back to the specified `redirect_uri`, a json error payload will be returned. As we don't process invalid callback urls.

Reply:
But you wrote 'correct' instead

Reply:
Yes - as the example below is a redirect - when client_id and redirect_uri params are correct, but has other errors (in the example wrong `response_type`).

```
< HTTP/1.1 302 Found
Found. Redirecting to https://your-app.example.com/callback?error=unsupported_response_type&error_description=only+supports%3A+code&state=dK6fRBNVY1HM2GQ4aBwM
```

The one or both client_id and redirect_uri is wrong, is on #45.